### PR TITLE
Add missing unit tests

### PR DESCRIPTION
**Pull request by adammahmood**
### What is the context of this PR?
- Adds missing unit tests highlighted by our test coverage

### How to review 
- Run tests

### CODE REVIEW

**Comment by adammahmood** [comment]
[Improve test coverage on Author](https://trello.com/c/vyPqDT9o/586-improve-test-coverage-on-author)

File: src/components/AnswerEditor/answereditor.test.js [hunk 1]
```diff
@@ -152,4 +164,20 @@ describe("Answer Editor", () => {
...
+      ...mockAnswer,
+      type: DATE,
+      properties: { format: "mm/yyyy" }
+    };
+    const instance = createWrapper({
+      answer,
+      ...mockMutations
+    }).instance();
+
+    expect(instance.formatIncludes("dd")(answer)).toEqual(false);
```
Comment (tgandrews) [inline src/components/AnswerEditor/answereditor.test.js]: Can we derive this another way as this is testing private API of the component not the expected output.

File: src/components/Answers/Date/index.test.js [hunk 2]
```diff
@@ -19,11 +19,10 @@ describe("Date", () => {
     handleUpdate = jest.fn();
     store = createMockStore();
```
Comment (tgandrews) [inline src/components/Answers/Date/index.test.js]: We can remove this as well.

File: src/components/AnswerEditor/answereditor.test.js (hunk 1 above)
Comment (adammahmood) [reply src/components/AnswerEditor/answereditor.test.js]: Not that I can think of.  Any ideas?

File: src/components/App/index.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      <div>Content</div>
+    </UnconnectedApp>
+  );
+
+describe("components/App", () => {
+  let props, wrapper;
+
+  beforeEach(() => {
+    props = {
+      gainConnection: jest.fn(),
```
Comment (tgandrews) [inline src/components/App/index.test.js]: We could stub the window to prove these are called.

File: src/components/MoveSectionModal/index.test.js [hunk 4]
```diff
@@ -1,25 +1,48 @@
...
-    );
+  let questionnaire, section, props, wrapper;
+
+  beforeEach(() => {
+    questionnaire = buildQuestionnaire();
+    section = questionnaire.sections[0];
+    props = {
+      questionnaire,
+      section,
+      onClose: jest.fn(),
```
Comment (tgandrews) [inline src/components/MoveSectionModal/index.test.js]: We could add a test to ensure this is called.

File: src/components/AnswerEditor/answereditor.test.js (hunk 1 above)
Comment (tgandrews) [reply src/components/AnswerEditor/answereditor.test.js]: They result of the functions are shown - so if the answer is type "Date" then you can see them in the output.

**Pull request merged**